### PR TITLE
[IOTDB-3778] Use SeriesScanUtil in compaction for point reader

### PR DESCRIPTION
**Pull request by MrQuansy**
In new MPP query framework, we create a new scan util called SeriesScanUtil and the old SeriesReader will be removed together with old standalone IoTDB.
However, current compaction still depends on the SeriesReader, we need to change it to depend on SeriesScanUtil instead.

### CODE REVIEW

**Comment by MrQuansy** [comment]
@liuminghui233 @choubenson @THUMarkLau PTAL~

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      boolean ascending) {
+    if (seriesPath instanceof AlignedPath) {
+      this.seriesScanUtil =
+          new AlignedSeriesScanUtil(
+              seriesPath, allSensors, context, timeFilter, valueFilter, ascending);
+    } else {
+      this.seriesScanUtil =
+          new SeriesScanUtil(
+              seriesPath, allSensors, dataType, context, timeFilter, valueFilter, ascending);
+    }
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java:63]:
```suggestion
   else if(seriesPath instanceof MeasurementPath) {
    XXXX
   } else {
     throw new IllegalArgumentException("XXXX");
   }
```

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+  public SeriesDataBlockReader(
+      PartialPath seriesPath,
+      Set<String> allSensors,
+      TSDataType dataType,
+      FragmentInstanceContext context,
+      QueryDataSource dataSource,
+      Filter timeFilter,
+      Filter valueFilter,
+      boolean ascending) {
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java:54]: For compaction, the parameters timeFilter & valueFilter are always null and the parameter ascending is always true. We can consider removing them in the constructor

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/fragment/FragmentInstanceContext.java [hunk 3]
```diff
@@ -72,13 +72,24 @@ public static FragmentInstanceContext createFragmentInstanceContext(
...
   private FragmentInstanceContext(
       FragmentInstanceId id, FragmentInstanceStateMachine stateMachine) {
     this.id = id;
     this.stateMachine = stateMachine;
     this.executionEndTime.set(END_TIME_INITIAL_VALUE);
   }
 
+  // used for compaction
+  private FragmentInstanceContext(long queryId) {
+    this.queryId = queryId;
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/fragment/FragmentInstanceContext.java:88]: `queryId` will be deprecated. Consider other ways to mock FragmentInstanceContext?

File: server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/source/SeriesScanUtil.java [hunk 4]
```diff
@@ -392,6 +401,7 @@ void skipCurrentChunk() {
    * overlapped pages are consumed
    */
   @SuppressWarnings("squid:S3776")
+  public
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/source/SeriesScanUtil.java:404]: It looks like the declaration of this method is separated by a comment.

File: server/src/main/java/org/apache/iotdb/db/query/context/QueryContext.java [hunk 5]
```diff
@@ -48,7 +48,7 @@ public class QueryContext {
    */
   private final Map<String, List<Modification>> fileModCache = new HashMap<>();
 
-  private long queryId;
+  protected long queryId;
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/query/context/QueryContext.java:51]: `queryId` will be deprecated.

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    if (seriesPath instanceof AlignedPath) {
+      this.seriesScanUtil =
+          new AlignedSeriesScanUtil(
+              seriesPath, allSensors, context, timeFilter, valueFilter, ascending);
+    } else {
```
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java:59]:
```suggestion
    } else if(seriesPath instanceof MeasurementPath) {
```

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java (hunk 1 above)
Comment (liuminghui233) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/reader/SeriesDataBlockReader.java:63]:
```suggestion
    } else {
    throw new IllegalArgumentException("");
    }
```

**Review by liuminghui233 — APPROVED** [review]
LGTM :-)

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**